toolchain/sem_ir/inst.h

```diff
@@ -19,20 +19,18 @@
 
 namespace Carbon::SemIR {
 
+// InstLikeTypeInfo is an implementation detail, and not public API.
+namespace Detail {
```

Comment:
We've called these `Internal` in other parts of the codebase, probably not worth diverging here.

There is a thread in Discord to see if that's the name we want to stick with or change to something else.

Reply:
Done.